### PR TITLE
Sync GNA lib calls to avoid multi threads and plugins crash

### PR DESCRIPTION
**Pull request by kbruniec**
Enables *-31708

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
can we check if this enables *-31708 or *-31709?

File: inference-engine/tests/functional/inference_engine/core_threading_tests.cpp [hunk 1]
```diff
@@ -118,7 +118,7 @@ TEST_F(CoreThreadingTests, RegisterPlugins) {
 
 // tested function: GetAvailableDevices, UnregisterPlugin
 // TODO: some plugins initialization (e.g. GNA) failed during such stress-test scenario
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/core_threading_tests.cpp:120]: can we remove this comment with TODO?

Comment (kbruniec) [reply inference-engine/tests/functional/inference_engine/core_threading_tests.cpp]: done

File: inference-engine/src/gna_plugin/gna_device.cpp [hunk 2]
```diff
@@ -62,6 +63,8 @@ uint32_t GNADeviceHelper::propagate(const intel_nnet_type_t *pNeuralNetwork,
     return reqId;
 }
 #else
+std::mutex GNADeviceHelper::acrossPluginsSync{};
```
Comment (ilya-lavrenov) [inline inference-engine/src/gna_plugin/gna_device.cpp:66]: should not it be marked as `static` or put into unnamed namespace?

Comment (kbruniec) [reply inference-engine/src/gna_plugin/gna_device.cpp:66]: Its declared as `static` class member so no need put `static` here in initialization.

Comment (ilya-lavrenov) [reply inference-engine/src/gna_plugin/gna_device.cpp:66]: oh, I see. I thought those are different variables..

**Comment by dorloff** [comment]
@kbruniec, why is it marked as draft?

**Review by dorloff — APPROVED** [review]

**Comment by kbruniec** [comment]
> 
> 
> @kbruniec, why is it marked as draft?
I was checking 31709. But sure, it can be merged now. Will all separate PR if needed.

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
Suggest handling of 31709 in a separate PR, @dorloff please, merge

**Pull request merged**